### PR TITLE
Score documents: Document selection and selection output

### PR DESCRIPTION
**Pull request by PrimozGodec**
##### Issue

Score documents widget is missing selection

##### Description of changes
Implemented selection in the table view and the `Selected documents` output

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/text/widgets/owscoredocuments.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 from typing import List, Callable, Tuple, Union
 
 import numpy as np
+from Orange.widgets.utils.annotated_data import create_annotated_table
```
Comment (VesnaT) [inline orangecontrib/text/widgets/owscoredocuments.py:7]: It would be nicer, if the imports were more structured.

**Review by VesnaT — COMMENTED** [review]
1. The selection use usually considered as a schema_only setting.
2. Have you considered adding a selection box. Something like:
image

**Comment by PrimozGodec** [comment]
> The selection use usually considered as a schema_only setting.

Totally forgot that it would be nice if the selection is remembered. Will implement it

> Have you considered adding a selection box. Something like:

I do not like selection boxes too much, since users can easily do selection in the table itself, but if other widgets have them I can also implement it here. 

> It would be nicer, if the imports were more structured.

I forgot to run isort on this one. Will fix it.

File: orangecontrib/text/widgets/owscoredocuments.py [hunk 2]
```diff
@@ -183,12 +210,20 @@ def callback(i: float) -> None:
         state.set_partial_result((sm, aggregation, scs))
 
 
+class SelectionMethods:
+    NONE, ALL, MANUAL, N_BEST = range(4)
+    ITEMS = "None", "All", "Manual", "Top words"
```
Comment (VesnaT) [inline orangecontrib/text/widgets/owscoredocuments.py:215]: Considering the widget scores the documents, `Top words` should be renamed to `Top documents`.

File: orangecontrib/text/widgets/owscoredocuments.py [hunk 3]
```diff
@@ -324,6 +369,31 @@ def _setup_control_area(self) -> None:
         gui.rubber(self.controlArea)
         gui.auto_send(self.buttonsArea, self, "auto_commit")
 
+        # select words box
+        box = gui.vBox(self.controlArea, "Select Words")
```
Comment (VesnaT) [inline orangecontrib/text/widgets/owscoredocuments.py:373]: `Select Words` -> `Select Documents` 
`self.controlArea` -> `self.buttonsArea`

**Review by VesnaT — CHANGES_REQUESTED** [review]

**Pull request merged**